app/src/components/LayoutHeader.vue

```diff
@@ -42,30 +42,138 @@
+        <div class="group-hover:text-grey-500">12</div>
+      </div>
+
+      <div class="border-r border-grey-100 h-14"></div>
+
+      <div class="group relative">
+        <div class="flex items-center h-16 gap-x-2 space-x-4 group cursor-pointer">
+          <div class="hidden md:block group-hover:text-grey-500">
+            <span>{{ userDisplayName }}</span>
+          </div>
+          <figure>
+            <Jazzicon :address="userAddress" :key="userAddress" width="40" />
+          </figure>
+          <div>
+            <ArrowBottomIcon class="icon-small icon-primary" />
+          </div>
+        </div>
+        <!-- menu-->
+        <div
+          class="
+            absolute
+            hidden
+            group-hover:flex
+            z-10
+            border border-grey-400
+            p-5
+            right-2
+            md:right-20
+            bg-white
+            text-grey-400
+            flex-col
+            gap-y-2
+            uppercase
+            font-medium
+            text-left
+            whitespace-nowrap
+          "
+        >
+          <div>{{ userDisplayName }}</div>
+          <div>{{ balance }} ETH</div>
+          <div class="border-b border-grey-400 my-4"></div>
+
+          <button
+            @click="changeWallet"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            change wallet
+          </button>
+          <button
+            @click="disconnect"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            disconnect wallet
+          </button>
+        </div>
       </div>
-    </nav>
+    </div>
+    <div v-else-if="!isSupportedNetwork" class="ml-auto flex items-center">
+      <div class="flex h-12 items-center pr-2 w-12">
+        <WarningIcon class="icon stroke-pink" />
+      </div>
+      <div class="text-gray-500">Unsupported network</div>
+    </div>
+    <div v-else @click="connectWallet" class="flex items-center h-14 gap-x-2 group cursor-pointer ml-auto">
+      <div class="hidden md:block group-hover:text-grey-500">Connect</div>
+      <div>
+        <ConnectWalletIcon class="icon icon-primary" />
+      </div>
+    </div>
   </header>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 import useWalletStore from 'src/store/wallet';
 import { WarningIcon } from '@fusion-icons/vue/interface';
+import { Wallet3Icon as ConnectWalletIcon } from '@fusion-icons/vue/web3';
+import { ArrowBottom2Icon as ArrowBottomIcon } from '@fusion-icons/vue/interface';
+import Jazzicon from 'src/components/Jazzicon.vue';
+import { Cart2Icon as CartIcon } from '@fusion-icons/vue/interface';
+import { formatEther, JsonRpcProvider, Web3Provider } from 'src/utils/ethers';
 
 // Header menu bar items
 const navigation = [
   { name: 'Home', href: '/' },
   { name: 'Cart', href: '/cart' },
-  { name: 'Contact', href: '/contact' },
 ];
```

Comment:
A bit of a nit, but to me href indicates a full URL, but here we just link to other parts of the app. Also IMO best practice for routing is to route by name, since that's just internal so we can ensure it never changes, whereas the route slug may change. You can find a route's `name` property defined in `app/src/router/index.ts` 

Therefore you'd have something like this:

```typescript
const navigation = [
  { name: 'Home', name: 'Home' },
  { name: 'Cart', name: 'Cart' },
];
```

More info: https://next.router.vuejs.org/guide/essentials/named-routes.html

Reply:
Makes total sense. Changed it! Using **label** and **name** properties now.

Reply:
Ah yes thanks for fixing my broken snippet above haha, `label` and `name` sounds great 👌